dipy/align/imaffine.py

```diff
@@ -922,25 +970,50 @@ def _init_optimizer(self, static, moving, transform, params0,
                 array, shape (dim+1, dim+1)
             If None:
                 Start from identity
+        static_mask : array, shape (S, R, C) or (R, C), optional
+            static image mask that defines which pixels in the static image
+            are used to calculate the mutual information.
+        moving_mask : array, shape (S', R', C') or (R', C'), optional
+            moving image mask that defines which pixels in the moving image
+            are used to calculate the mutual information.
 
         """
         self.dim = len(static.shape)
         self.transform = transform
         n = transform.get_number_of_parameters()
         self.nparams = n
 
+        # ensure that masks are not all zeros
+        if np.all(static_mask == 0):
+            warn("static_mask is all zeros, setting to None", UserWarning)
```

Comment:
This means that the entire volume will be used, right? I thiink we can say that explicitly 

```suggestion
            warn("static_mask is all zeros, setting to None (which means that the entire volume will be used)", UserWarning)
```